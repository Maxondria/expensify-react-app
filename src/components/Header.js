import React from "react";
import { NavLink } from "react-router-dom";
import { startLogout } from "../redux/actions/auth";
import { connect } from "react-redux";

export const Header = props => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active">
      Home
    </NavLink>
    {" | "}
    <NavLink activeClassName="is-active" to="/create">
      Add Expense
    </NavLink>
    {" | "}
    <button onClick={props.startLogout}>Logout </button>
  </header>
);

const matchDispatchToProps = {
  startLogout
};

export default connect(
  undefined,
  matchDispatchToProps
)(Header);
