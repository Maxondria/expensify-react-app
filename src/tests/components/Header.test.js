import ReactShallowRenderer from "react-test-renderer/shallow";
import React from "react";
import Header from "../../components/Header";

it("should render Header correctly", function() {
  const renderer = new ReactShallowRenderer();

  renderer.render(<Header />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
