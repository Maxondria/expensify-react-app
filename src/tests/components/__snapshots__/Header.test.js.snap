// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <ForwardRef
    activeClassName="is-active"
    to="/"
  >
    Home
  </ForwardRef>
   | 
  <ForwardRef
    activeClassName="is-active"
    to="/create"
  >
    Add Expense
  </ForwardRef>
   | 
  <ForwardRef
    activeClassName="is-active"
    to="/help"
  >
    Help
  </ForwardRef>
</header>
`;
